usecases/objects/auto_schema.go

```diff
@@ -492,3 +492,73 @@ func (m *autoSchemaManager) determineNestedPropertiesOfArray(valArray []interfac
 
 	return nestedProperties, nil
 }
+
+type classTenants map[*models.Class]map[string]struct{}
+
+func (m *autoSchemaManager) autoTenants(ctx context.Context,
+	principal *models.Principal, objects []*models.Object,
+) error {
+	ct := make(classTenants)
+
+	for _, obj := range objects {
+		// TODO: track the classes we have already checked
+		//       before calling getClass again
+		o := models.Object{Class: obj.Class}
+		class, err := m.getClass(principal, &o)
+		if err != nil {
+			return err
+		}
+
+		// class not found in schema. this will have already
+		// been caught by b.validateObjectsConcurrently, and
+		// will be handled accordingly, so we don't need to
+		// do anything here
+		if class == nil {
+			continue
+		}
+
+		if shouldAutoCreateTenants(class) {
+			cls, ok := ct[class]
+			if !ok {
+				ct[class] = map[string]struct{}{obj.Tenant: {}}
+				continue
+			}
+			cls[obj.Tenant] = struct{}{}
+		}
+	}
+
+	for c, ts := range ct {
+		l := make([]*models.Tenant, len(ts))
+		i := 0
+		for t := range ts {
+			l[i] = &models.Tenant{Name: t}
+			i++
+		}
+		if err := m.addTenants(ctx, principal, c, l); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (m *autoSchemaManager) addTenants(ctx context.Context, principal *models.Principal,
+	class *models.Class, tenants []*models.Tenant,
+) error {
+	if len(tenants) == 0 {
+		return fmt.Errorf(
+			"tenants must be included for multitenant-enabled class %q", class.Class)
+	}
+	if err := m.schemaManager.AddTenants(ctx, principal, class.Class, tenants); err != nil {
+		if !strings.Contains(err.Error(), "already exists") {
```

Comment:
Actually, this check is not even needed, since the `GetPartitions` method just skips any shards (tenants) which already exist, without yielding an error. Removed